internal/cli/atlas/networking/peering/watch.go

```diff
@@ -57,28 +57,28 @@ func (opts *WatchOpts) watcher() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-
-	switch v := result.(type) {
-	case *atlasv2.AwsNetworkPeeringConnectionSettings:
-		return watcherAWS(v), nil
-	case *atlasv2.AzureNetworkPeeringConnectionSettings:
-		return watcherAzure(v), nil
-	case *atlasv2.GCPNetworkPeeringConnectionSettings:
-		return watcherGCP(v), nil
+	switch v := result.GetProviderName(); v {
+	case "AWS":
+		return watcherAWS(result), nil
+	case "AZURE":
+		return watcherAzure(result), nil
+	case "GCP":
+		return watcherGCP(result), nil
```

Comment:
the swtch was to allow casting to the actual instance, is this switch still necessary? feels like no

Reply:
Do you mean replacing the switch with ifs?
GolangCiLint forces us to use a switch for more than 2 If/else :(
I also investigated if we can ignore provider details in this case but it did generate invalid behaviour for GCP.

Reply:
watcherAWS, watcherAzure and watcherGCP( are all now the same method, I don't think we need this switch or three methods that do the same

Reply:
on second look only azure and gcp are the same, aws is slightly different, so we can combine azure and gcp and move from a switch to an if but the change of the switch is not necessary, it was before if the methods were the same

Reply:
yes. This is the tradeof between verbosity and type clarity. If we merge both

Reply:
I would prefer to not address it as it would affect support for future provider changes.